service-base/src/main/java/org/eclipse/hono/service/metric/MicrometerBasedMetrics.java

```diff
@@ -110,8 +117,14 @@ public final void incrementConnections(final String tenantId) {
     public final void decrementConnections(final String tenantId) {
 
         Objects.requireNonNull(tenantId);
-        gaugeForTenant(METER_CONNECTIONS_AUTHENTICATED, this.authenticatedConnections, tenantId, AtomicLong::new)
+        final long connections = gaugeForTenant(METER_CONNECTIONS_AUTHENTICATED, this.authenticatedConnections,
```

Comment:
Are we sure that this code is thread safe?

Reply:
IMHO the trigger (connection count for the tenant drops to 0) is a problem and parallel connects and disconnects are quite likely to happen. The value can change quite often between two scraping intervals and can only be removed if it is 0 for a longer time. So this needs indeed to be changed. The question remains what a better trigger event could be.

Reply:
I think you can only solve this properly by synchronizing the increment and decrement operation.